utils/compose/builder.go

```diff
@@ -39,6 +39,10 @@ func (b *Builder) AddService(sb ServiceBuilder) *Builder {
 	s := sb.Service()
 
 	for i, v := range s.Volumes {
+		nv, _, ok := strings.Cut(v, ":")
+		if ok && b.c.Volumes[nv] != nil {
+			continue
+		}
```

Comment:
I don't see it. `Compose.AddVolume` saves whole string, but here you cut it, why?

Reply:
This is to support named data container like in the [following example](https://docs.docker.com/compose/compose-file/07-volumes/#example).